src/main/kotlin/com/cognifide/gradle/aem/pkg/deploy/HtmlResponse.kt

```diff
@@ -49,5 +52,48 @@ abstract class HtmlResponse(private val rawHtml: String) {
 
     val success: Boolean
         get() = status == Status.SUCCESS
+
+    companion object {
+        private const val NUMBER_OF_LINES_TO_READ = 5000
+
+        private const val ERROR_SEPARATOR = "\n\n"
+
+        private const val LINE_FEED = 10.toChar()
+
+        fun readFrom(stream: InputStream, errorPatterns: List<ErrorPattern>, statusTags: List<String>): String {
+            val buf = IOUtils.toBufferedInputStream(stream)
+            val reader = buf.bufferedReader()
```

Comment:
why `buf` is introduced? I dislike such extra vars

Reply:
when `readFrom` and `readBylines` split have no make sense for me; extracted stream buffering gives nothing here because these 2 methods have sense only when they are combined / have single responsibility.